### PR TITLE
feat: otel tracing

### PR DESCRIPTION
**Pull request by fenos**
## What kind of change does this PR introduce?

Feature

## What is the new behavior?

This PR introduces optional OTEL traces

### CODE REVIEW

File: migrations/multitenant/0010-delete-cache-cache-notifier-on-delete.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+CREATE FUNCTION tenants_delete_notify_trigger ()
+    RETURNS TRIGGER
+AS $$
+BEGIN
+    PERFORM
+        pg_notify('tenants_update', '"' || OLD.id || '"');
```
Comment (inian) [inline migrations/multitenant/0010-delete-cache-cache-notifier-on-delete.sql:6]: what is this used for?

File: monitoring/otel/config/otel-collector-config.yml [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                    },
+                },
+                {
+                  # trace.mode = premium
+                  name: trace-mode-default,
+                  type: string_attribute,
+                  string_attribute:
+                    {
+                      key: trace.mode,
+                      values: [ premium ],
```
Comment (inian) [inline monitoring/otel/config/otel-collector-config.yml]: premium is a weird term for this. enhanced?

**Review by inian — APPROVED** [review]

File: migrations/multitenant/0010-delete-cache-cache-notifier-on-delete.sql (hunk 1 above)
Comment (fenos) [reply migrations/multitenant/0010-delete-cache-cache-notifier-on-delete.sql:6]: When a DELETE request to tenants is issued, we currently don't clear the cache across all storage-api services. 
Only on the instance that received the DELETE request.

By using this trigger on delete all containers will have a consistent view of the tenants cache

File: monitoring/otel/config/otel-collector-config.yml (hunk 2 above)
Comment (fenos) [reply monitoring/otel/config/otel-collector-config.yml]: Sure, can change to `enhanced` or `always_trace`

File: monitoring/otel/config/otel-collector-config.yml [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                    },
+                },
+                {
+                  name: success-status-codes,
+                  type: numeric_attribute,
+                  numeric_attribute: { key: http.status_code, min_value: 200, max_value: 399 }
+                },
+                {
+                  name: basic-sampling,
```
Comment (inian) [inline monitoring/otel/config/otel-collector-config.yml]: should the name be changed here too?

**Pull request merged**